src/Symfony/Component/Console/Color.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console;
+
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+final class Color
+{
+    private static $colors = [
+        'black' => 0,
+        'red' => 1,
+        'green' => 2,
+        'yellow' => 3,
+        'blue' => 4,
+        'magenta' => 5,
+        'cyan' => 6,
+        'white' => 7,
+        'default' => 9,
+    ];
+
+    private static $availableOptions = [
+        'bold' => ['set' => 1, 'unset' => 22],
+        'underscore' => ['set' => 4, 'unset' => 24],
+        'blink' => ['set' => 5, 'unset' => 25],
+        'reverse' => ['set' => 7, 'unset' => 27],
+        'conceal' => ['set' => 8, 'unset' => 28],
+    ];
+
+    private $foreground;
+    private $background;
+    private $options = [];
+
+    public function __construct(string $foreground = '', string $background = '', array $options = [])
+    {
+        $this->foreground = $this->parseColor($foreground);
+        $this->background = $this->parseColor($background);
+
+        foreach ($options as $option) {
+            if (!isset(static::$availableOptions[$option])) {
```

Comment:
should be `self::` to access a private static state, not `static::` (as the class is final, this won't be possible to actually break the code for that, as it happens when extending the class, but still not a good idea IMO)

Reply:
btw, it looks like the existing code i `OutputFormatterStyle` also suffers from this. And there, it can actually create issues as the class is not final.

Reply:
fixed (for OutputFormatterStyle, another PR should be created).